js/skin.js

```diff
@@ -287,8 +287,42 @@ var Skin = createReactClass({
     }
   },
 
+  /**
+   * call handleTouchEnd when touchend was called on selectedScreen
+   * @public
+   * @param {Event} event - event object
+   */
+  handleTouchEnd: function(event) {
+    event.preventDefault(); // to prevent mobile from propagating click to discovery shown on pause
+    if (this.props.controller.state.controlBarVisible) {
+      let shouldToggle = false;
+      if (this.props.controller.videoVr) {
+        if (!this.state.isVrMouseMove) {
+          shouldToggle = true;
+        }
+      } else {
+        shouldToggle = true;
+      }
+      if (shouldToggle) {
+        this.props.controller.togglePlayPause(event);
+      }
+      // set value for this.state.isVrMouseMove in this.handleVrPlayerMouseUp
+    }
+    if (this.props.controller.videoVr) { // only for vr on mobile
+      // Check current a vr video position (an user could change position using tilting)
+      if (
+        typeof this.props.controller.checkVrDirection === 'function' &&
+        typeof this.props.controller.setControllerVrViewingDiraction === 'function'
```

Comment:
We are owners of the controller.js, so it's a bit unclear why do we check if we didn't forget to implement method of our controller? Consider we should trust ourselves.

Reply:
we need to check it according by style docs.

Reply:
:-)

Reply:
I agree with @zelibobla , I think for our own methods we probably only need to check in cases in which it's an optional prop that might've not been passed.